.ci/ci-fast-return.sh

```diff
@@ -0,0 +1,183 @@
+
+	# Use just the repo name itself. The YAML does not like having the full
+	# repo path in it.
+	local repo="${kata_repo##*/}"
+	if [ -n "$repo" ]; then
+		local repo_skip=$(yq read ${yqfile} ${repo}.skip_patterns)
+		[ "$repo_skip" == "null" ] && repo_skip=""
+		local repo_check=$(yq read ${yqfile} ${repo}.check_patterns)
+		[ "$repo_check" == "null" ] && repo_check=""
+	else
+		info "No repo set, skipping repo specific patterns" >&2
+	fi
+
+	local canskip=""
+	for x in $common_skip $repo_skip; do
+		canskip+=$(echo "-e $x ")
+	done
+
+	local mustcheck=""
+	for x in $common_check $repo_check; do
+		mustcheck+=$(echo "-e $x ")
+	done
+
+	info "Skips [$canskip]" >&2
+	info "Checks [$mustcheck]" >&2
+
+	# do we have any patterns to check?
+	if [ -n "$mustcheck" ]; then
+		local need_checking=$(egrep ${mustcheck[@]} <<< "$files" || true)
+	else
+		info "No force CI check patterns" >&2
+		local need_checking=""
+	fi
+
+	# If we have any files that *must* be checked, then immediately return
+	if [ -n "$need_checking" ]; then
+		# stderr, so it does not get in our return value...
+		cat >&2 <<-EOT
+		INFO: Cannot fastpath skip CI.
+		INFO: Some files present must be CI checked.
+		INFO: Files to check are:
+
+		$need_checking
+
+EOT
+		echo "1"
+		return
+	else
+		info "No force check files found" >&2
+	fi
+
+	# Now we have checked there are no 'must check' files, we can check to see
+	# if all files fall into the 'can skip' patterns.
+	# first, do we have any skip patterns to search for?
+	if [ -n "$canskip" ]; then
+		local non_skippable=$(egrep -v ${canskip[@]} <<< "$files" || true)
+	else
+		info "No skip CI check patterns" >&2
+		# No patterns, set non-skippable list to all files then...
+		local non_skippable="$files"
+	fi
+
+	if [ -z "$non_skippable" ]; then
+		# stderr, so it does not get in our return value...
+		cat >&2 <<-EOT
+		INFO: No files to check in CI.
+		INFO: Fastpath short circuit returning from CI.
+		INFO: Files skipped are:
+
+		$files
+
+EOT
+		echo "0"
+		return
+	else
+		cat >&2 <<-EOT
+		INFO: Not all files skippable
+
+		$non_skippable
+
+EOT
+		echo "1"
+		return
+	fi
+}
+
+# Check if we have the 'magic label' that forces a CI run set on the PR
+# Return:
+#  0 - No label found, could skip the CI
+#  1 - Label found - should run the CI
+check_force_label() {
+
+	[ -z "$ghprbGhRepository" ] && info "No ghprbGhRepository set, skip label check" >&2 && echo "0" && return
+	[ -z "$ghprbPullId" ] && info "No ghprbPullId set, skip label check" >&2 && echo "0" && return
+
+	info "Checking labels for PR ${ghprbGhRepository}/${ghprbPullId}" >&2
+
+	# Pull the label list for the PR
+	# Ideally we'd use a github auth token here so we don't get rate limited, but to do that we would
+	# have to expose the token into the CI scripts, which is then potentially a security hole.
+	# json=$(curl -H "Authorization: Basic ${GITHUB_API_TOKEN}" https://api.github.com/repos/${ghprbGhRepository}/issues/${ghprbPullId}/labels)
```

Comment:
I'd remove this and maybe slap it into a feature issue in case we find a way to do this later.